modules/rostests/kmtests/ntos_ob/ObInsert.c

```diff
@@ -0,0 +1,651 @@
+ * PURPOSE:         Kernel-Mode Test Suite - ObInsertObject test
+ * PROGRAMMER:      Gleb Surikov <glebs.surikovs@gmail.com>
+ */
+
+#include <kmt_test.h>
+
+#define NDEBUG
+#include <debug.h>
+
+typedef struct DUMMY_TYPE
+{
+    ULONG Data;
+} DUMMY_TYPE, *PDUMMY_TYPE;
+
+static HANDLE DirectoryHandle;
+static UNICODE_STRING DirectoryName;
+
+static POBJECT_TYPE DummyType;
+static UNICODE_STRING DummyTypeName;
+static OBJECT_TYPE_INITIALIZER ObjectTypeInitializer;
+static OBJECT_ATTRIBUTES DummyObjectAttributes;
+
+static
+VOID
+NTAPI
+DumpProc(
+    IN PVOID Object,
+    IN POB_DUMP_CONTROL DumpControl
+)
+{
+    DPRINT("DumpProc() called\n");
+}
+
+static
+NTSTATUS
+NTAPI
+OpenProc(
+    IN OB_OPEN_REASON OpenReason,
+    IN PEPROCESS Process,
+    IN PVOID Object,
+    IN ACCESS_MASK GrantedAccess,
+    IN ULONG HandleCount
+)
+{
+    DPRINT("OpenProc() 0x%p, OpenReason %d, HandleCount %lu, AccessMask 0x%lX\n",
+           Object,
+           OpenReason,
+           HandleCount,
+           GrantedAccess);
+    return STATUS_SUCCESS;
+}
+
+static
+VOID
+NTAPI
+CloseProc(
+    IN PEPROCESS Process,
+    IN PVOID Object,
+    IN ACCESS_MASK GrantedAccess,
+    IN ULONG ProcessHandleCount,
+    IN ULONG SystemHandleCount
+)
+{
+    DPRINT("CloseProc() 0x%p, ProcessHandleCount %lu, SystemHandleCount %lu, AccessMask 0x%lX\n",
+           Object,
+           ProcessHandleCount,
+           SystemHandleCount,
+           GrantedAccess);
+}
+
+static
+VOID
+NTAPI
+DeleteProc(IN PVOID Object)
+{
+    DPRINT("DeleteProc() 0x%p\n", Object);
+}
+
+static
+NTSTATUS
+NTAPI
+ParseProc(
+    IN PVOID ParseObject,
+    IN PVOID ObjectType,
+    IN OUT PACCESS_STATE AccessState,
+    IN KPROCESSOR_MODE AccessMode,
+    IN ULONG Attributes,
+    IN OUT PUNICODE_STRING CompleteName,
+    IN OUT PUNICODE_STRING RemainingName,
+    IN OUT PVOID Context OPTIONAL,
+    IN PSECURITY_QUALITY_OF_SERVICE SecurityQos OPTIONAL,
+    OUT PVOID *Object
+)
+{
+    DPRINT("ParseProc() called\n");
+    *Object = NULL;
+    return STATUS_OBJECT_NAME_NOT_FOUND;
+}
+
+static
+BOOLEAN
+NTAPI
+OkayToCloseProc(
+    IN PEPROCESS Process OPTIONAL,
+    IN PVOID Object,
+    IN HANDLE Handle,
+    IN KPROCESSOR_MODE AccessMode
+)
+{
+    DPRINT("OkayToCloseProc() 0x%p, Handle 0x%p, AccessMask 0x%lX\n",
+           Object,
+           Handle,
+           AccessMode);
+    return TRUE;
+}
+
+static
+NTSTATUS
+NTAPI
+QueryNameProc(
+    IN PVOID Object,
+    IN BOOLEAN HasObjectName,
+    OUT POBJECT_NAME_INFORMATION ObjectNameInfo,
+    IN ULONG Length,
+    OUT PULONG ReturnLength,
+    IN KPROCESSOR_MODE AccessMode
+)
+{
+    DPRINT("QueryNameProc() 0x%p, HasObjectName %d, Len %lu, AccessMask 0x%lX\n",
+           Object,
+           HasObjectName,
+           Length,
+           AccessMode);
+
+    ObjectNameInfo = NULL;
+    ReturnLength = 0;
+    return STATUS_SUCCESS;
+}
+
+/*!
+ * @brief This function creates a test directory.
+ */
+static
+VOID
+ObInsert_CreateTestDirectory()
```

Comment:
```suggestion
ObInsert_CreateTestDirectory(VOID)
```
and same comment for all the other functions down below.
https://stackoverflow.com/a/693794/21852502
https://stackoverflow.com/questions/1163879/in-c-funcvoid-vs-func

Reply:
Thanks!